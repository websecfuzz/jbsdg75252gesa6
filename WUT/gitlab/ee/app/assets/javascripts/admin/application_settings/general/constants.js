import { s__ } from '~/locale';

export const DROPZONE_DESCRIPTION_TEXT = s__(
  'Licenses|Drag your license file here or %{linkStart}select a file to upload%{linkEnd}.',
);
export const FILE_UPLOAD_ERROR_MESSAGE = s__('Licenses|The file could not be uploaded.');
export const FILE_DROP_ERROR_MESSAGE = s__(
  'Licenses|Error: You are trying to upload something other than a file',
);
export const DROP_TO_START_MESSAGE = s__('Licenses|Drop your license file to start the upload.');
