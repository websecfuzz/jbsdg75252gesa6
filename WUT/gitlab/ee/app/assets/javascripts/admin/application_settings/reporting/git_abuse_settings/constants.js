import { s__, __, sprintf } from '~/locale';

export const MIN_NUM_REPOS = 0;
export const MAX_NUM_REPOS = 10000;

export const MIN_TIME_PERIOD = 0;
export const MAX_TIME_PERIOD = 10 * 24 * 60 * 60; // 10 days in seconds;

export const MAX_ALLOWED_USERS = 100;

export const MIN_ALERTED_USERS = 1;
export const MAX_ALERTED_USERS = 100;

export const NUM_REPOS_BLANK_ERROR = s__(
  "GitAbuse|Number of repositories can't be blank. Set to 0 for no limit.",
);
export const NUM_REPOS_NAN_ERROR = s__('GitAbuse|Number of repositories must be a number.');
export const NUM_REPOS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Number of repositories should be between %{minNumRepos}-%{maxNumRepos}.`),
  {
    minNumRepos: MIN_NUM_REPOS,
    maxNumRepos: MAX_NUM_REPOS,
  },
);

export const TIME_PERIOD_BLANK_ERROR = s__(
  "GitAbuse|Reporting time period can't be blank. Set to 0 for no limit.",
);
export const TIME_PERIOD_NAN_ERROR = s__('GitAbuse|Reporting time period must be a number.');
export const TIME_PERIOD_LIMIT_ERROR = sprintf(
  s__(
    `GitAbuse|Reporting time period should be between %{minTimePeriod}-%{maxTimePeriod} seconds.`,
  ),
  {
    minTimePeriod: MIN_TIME_PERIOD,
    maxTimePeriod: MAX_TIME_PERIOD,
  },
);

export const ALLOWED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|You cannot specify more than %{maxAllowedUsers} excluded users.`),
  {
    maxAllowedUsers: MAX_ALLOWED_USERS,
  },
);

export const NUM_REPO_LABEL = s__('GitAbuse|Number of repositories');
export const NUM_REPO_DESCRIPTION = s__(
  "GitAbuse|The maximum number of unique repositories a user can download in the specified time period before they're banned.",
);
export const REPORTING_TIME_PERIOD_LABEL = s__('GitAbuse|Reporting time period (seconds)');

export const ALLOWED_USERS_LABEL = s__('GitAbuse|Excluded users');
export const ALLOWED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are excluded from the Git abuse rate limit.',
);

export const ALERTED_USERS_LABEL = s__('GitAbuse|Send notifications to');
export const ALERTED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are emailed when Git abuse rate limit is exceeded.',
);
export const ALERTED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Select between %{minAlertedUsers} and %{maxAlertedUsers} users to notify.`),
  {
    minAlertedUsers: MIN_ALERTED_USERS,
    maxAlertedUsers: MAX_ALERTED_USERS,
  },
);

export const AUTO_BAN_TOGGLE_LABEL = s__(
  'GitAbuse|Automatically ban users from this %{scope} when they exceed the specified limits',
);

export const SEARCH_USERS = __('Search users');
export const SEARCH_TERM_TOO_SHORT = __('Enter at least three characters to search');
export const NO_RESULTS = __('No results');

export const SAVE_CHANGES = __('Save changes');
export const SUCCESS_MESSAGE = __('Application settings saved successfully.');

export const LOAD_ERROR_MESSAGE = __(
  'An error occurred while saving your settings. Try saving them again.',
);
export const SAVE_ERROR_MESSAGE = __(
  'An error occurred while retrieving your settings. Reload the page to try again.',
);
