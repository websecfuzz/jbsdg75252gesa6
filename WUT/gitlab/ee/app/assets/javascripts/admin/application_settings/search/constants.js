import { __, s__ } from '~/locale';

export const MINIMUM_QUERY_LENGTH = 3;
export const NO_RESULTS_TEXT = __('No results found.');
export const SEARCH_QUERY_TOO_SHORT = __('Enter at least three characters to search.');
export const ENTITIES_FETCH_ERROR = s__(
  'AdvancedSearch|Could not fetch index entities. Please try again later.',
);
