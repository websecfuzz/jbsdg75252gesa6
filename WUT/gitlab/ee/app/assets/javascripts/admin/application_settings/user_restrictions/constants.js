import { helpPagePath } from '~/helpers/help_page_helper';

export const PRIVATE_PROFILES_DISABLED_ICON = 'private-profiles-disabled-icon';
export const PRIVATE_PROFILES_DISABLED_HELP_LINK = helpPagePath(
  'administration/settings/account_and_limit_settings',
  { anchor: 'set-profiles-of-new-users-to-private-by-default' },
);
