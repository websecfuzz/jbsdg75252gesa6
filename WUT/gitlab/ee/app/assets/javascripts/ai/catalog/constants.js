import { __ } from '~/locale';

export const AI_CATALOG_ENUM_AGENT = 'AGENT';
export const AI_CATALOG_NAME_AGENT = __('agent');

export const TYPENAME_AI_CATALOG_ITEM = 'Ai::Catalog::Item';

export const ENUM_TO_NAME_MAP = {
  [AI_CATALOG_ENUM_AGENT]: AI_CATALOG_NAME_AGENT,
};

// Matches backend validations in https://gitlab.com/gitlab-org/gitlab/blob/aa02c3080b316cf0f3b71a992bc5cc5dc8e8bb34/ee/app/models/ai/catalog/item.rb#L10
export const MAX_LENGTH_NAME = 255;
export const MAX_LENGTH_DESCRIPTION = 1024;
export const MAX_LENGTH_PROMPT = 1000000;
