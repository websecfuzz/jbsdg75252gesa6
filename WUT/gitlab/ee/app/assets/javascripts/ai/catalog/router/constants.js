export const AI_CATALOG_INDEX_ROUTE = 'ai-catalog';
export const AI_CATALOG_AGENTS_ROUTE = '/agents';
export const AI_CATALOG_AGENTS_SHOW_ROUTE = '/agents/:id';
export const AI_CATALOG_AGENTS_RUN_ROUTE = '/agents/:id/run';
export const AI_CATALOG_AGENTS_NEW_ROUTE = '/agents/new';
