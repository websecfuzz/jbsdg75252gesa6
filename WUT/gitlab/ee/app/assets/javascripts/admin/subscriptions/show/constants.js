import { __, s__ } from '~/locale';
import { PROMO_URL } from '~/constants';

export const subscriptionMainTitle = s__('SuperSonics|Instance subscription');
export const subscriptionActivationNotificationText = s__(
  `SuperSonics|Your subscription was successfully activated. You can see the details below.`,
);
export const subscriptionActivationFutureDatedNotificationTitle = s__(
  'SuperSonics|Your future dated license was successfully added',
);
export const subscriptionActivationFutureDatedNotificationMessage = s__(
  'SuperSonics|You have successfully added a license that activates on %{date}. Please see the subscription history table below for more details.',
);
export const subscriptionActivationInsertCode = __(
  'Find the activation code for a purchased or renewed subscription in the Customers Portal or purchase confirmation email.',
);
export const howToActivateSubscription = s__(
  'SuperSonics|Learn how to %{linkStart}activate your subscription%{linkEnd}.',
);
export const subscriptionHistoryFailedTitle = s__('SuperSonics|Subscription unavailable');
export const subscriptionHistoryFailedMessage = s__(
  'SuperSonics|Your %{subscriptionEntryName} cannot be displayed at the moment. Please refresh the page to try again.',
);
export const currentSubscriptionsEntryName = s__('SuperSonics|current subscription');
export const pastSubscriptionsEntryName = s__('SuperSonics|past subscriptions');
export const futureSubscriptionsEntryName = s__('SuperSonics|future subscriptions');

export const cancelLabel = __('Cancel');
export const activateLabel = s__('AdminUsers|Activate');
export const activateSubscription = s__('SuperSonics|Activate subscription');
export const addActivationCode = s__('SuperSonics|Add activation code');
export const noActiveSubscription = s__(`SuperSonics|You do not have an active subscription`);
export const subscriptionDetailsHeaderText = s__('SuperSonics|Subscription details');
export const licensedToHeaderText = s__('SuperSonics|Licensed to');
export const copySubscriptionIdButtonText = __('Copy');
export const licenseFileText = __('Legacy license');
export const onlineCloudLicenseText = s__('SuperSonics|Online license');
export const offlineCloudLicenseText = s__('SuperSonics|Offline license');
export const detailsLabels = {
  address: __('Address'),
  company: __('Company'),
  email: __('Email'),
  id: __('ID'),
  lastSync: s__('Subscriptions|Last sync'),
  name: licensedToHeaderText,
  plan: __('Plan'),
  type: __('Type'),
  expiresAt: s__('Subscriptions|End date'),
  startsAt: s__('Subscriptions|Start date'),
};

export const subscriptionTable = {
  activatedAt: s__('Subscriptions|Activation date'),
  expiresOn: s__('Subscriptions|End date'),
  seats: __('Seats'),
  startsAt: s__('Subscriptions|Start date'),
  title: __('Subscription History'),
  type: __('Type'),
};

export const subscriptionActivationForm = {
  activationCode: s__('SuperSonics|Activation code'),
  activationCodeFeedback: s__(
    'SuperSonics|The activation code should be a 24-character alphanumeric string',
  ),
  pasteActivationCode: s__('SuperSonics|Enter activation code'),
  acceptTerms: s__(
    'SuperSonics|I agree that my use of the GitLab Software is subject to the Subscription Agreement located at the %{linkStart}Terms of Service%{linkEnd}, unless otherwise agreed to in writing with GitLab.',
  ),
  acceptTermsFeedback: s__('SuperSonics|Select that you agree before activating the subscription.'),
};

export const subscriptionSyncStatus = {
  SYNC_FAILURE: 'SYNC_FAILURE',
  SYNC_PENDING: 'SYNC_PENDING',
  SYNC_SUCCESS: 'SYNC_SUCCESS',
};

export const subscriptionTypes = {
  ONLINE_CLOUD: 'online_cloud',
  OFFLINE_CLOUD: 'offline_cloud',
  LEGACY_LICENSE: 'legacy_license',
};

export const trialCard = {
  title: s__('SuperSonics|Free trial'),
  description: s__(
    'SuperSonics|You can start a free trial of GitLab Ultimate without any obligation or payment details.',
  ),
  startTrial: s__('SuperSonics|Start free trial'),
};

export const buySubscriptionCard = {
  title: __('Subscription'),
  description: s__(
    'SuperSonics|Ready to get started? A GitLab plan is ideal for scaling organizations and for multi team usage.',
  ),
  buttonLabel: s__('SuperSonics|Buy subscription'),
};

export const SUBSCRIPTION_ACTIVATION_FAILURE_EVENT = 'subscription-activation-failure';
export const SUBSCRIPTION_ACTIVATION_SUCCESS_EVENT = 'subscription-activation-success';
export const SUBSCRIPTION_ACTIVATION_FINALIZED_EVENT = 'subscription-activation-finalized';

// Server-sent error messages (or regexes to match those messages) to detect error type
export const INVALID_ACTIVATION_CODE_SERVER_ERROR = 'invalid activation code';
export const EXPIRED_LICENSE_SERVER_ERROR = 'This license has already expired.';
export const SILENT_MODE_ENABLED_ERROR =
  'only get, head, options, and trace methods are allowed in silent mode';
export const SUBSCRIPTION_NOT_FOUND_SERVER_ERROR = 'without cloud compatible subscription';
export const SUBSCRIPTION_OVERAGES_SERVER_ERROR_REGEX =
  /(This GitLab installation currently has|During the year before this license started, this GitLab installation had) (\d+) active users?, exceeding this license's limit of (\d+) by (\d+) users?\. Please add a license for at least (\d+) users? or contact sales at https:\/\/about\.gitlab\.com\/sales\//;
export const SUBSCRIPTION_INSUFFICIENT_TRUE_UP_SERVER_ERROR_REGEX =
  /You have applied a True-up for (\d+) users? but you need one for (\d+) users?\. Please contact sales at https:\/\/about\.gitlab\.com\/sales\//;

export const CONNECTIVITY_ERROR = 'CONNECTIVITY_ERROR';
export const INVALID_CODE_ERROR = 'INVALID_CODE_ERROR';

export const supportLink = `${PROMO_URL}/support/#contact-support`;
export const exportLicenseUsageBtnText = s__('SuperSonics|Export license usage file');
export const customersPortalBtnText = s__('SuperSonics|Customers Portal');

export const instanceHasFutureLicenseBanner = {
  title: s__('SuperSonics|You have a future dated license'),
  message: s__(
    'SuperSonics|You have added a license that activates on %{date}. Please see the subscription history table below for more details.',
  ),
};

// Subscription Sync Button
export const SYNC_BUTTON_ID = 'syncButton';
export const syncButtonTexts = Object.freeze({
  syncSubscriptionButtonText: s__('SuperSonics|Sync subscription details'),
  syncSubscriptionTooltipText: s__(
    'SuperSonics|You can sync your subscription data to ensure your details are up to date.',
  ),
});

export const VIEW_ADMIN_SUBSCRIPTION_PAGELOAD = 'view_admin_subscription_pageload';
