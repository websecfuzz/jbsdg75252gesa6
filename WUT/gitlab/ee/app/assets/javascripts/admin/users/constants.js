import { s__ } from '~/locale';
import { getTokenConfigs } from '~/admin/users/constants';

export {
  SOLO_OWNED_ORGANIZATIONS_EMPTY,
  I18N_USER_ACTIONS,
  SOLO_OWNED_ORGANIZATIONS_REQUESTED_COUNT,
} from '~/admin/users/constants';

export const TOKEN_CONFIGS = getTokenConfigs([
  { value: 'admins', title: s__('AdminUsers|Administrator') },
  { value: 'auditors', title: s__('AdminUsers|Auditor') },
  { value: 'external', title: s__('AdminUsers|External') },
]);
