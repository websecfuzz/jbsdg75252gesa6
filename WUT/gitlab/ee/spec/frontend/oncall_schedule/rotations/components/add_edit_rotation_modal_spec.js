import { GlAlert, GlModal } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import VueApollo from 'vue-apollo';
import Vue, { nextTick } from 'vue';
import { cloneDeep } from 'lodash';
import AddEditRotationForm from 'ee/oncall_schedules/components/rotations/components/add_edit_rotation_form.vue';
import AddEditRotationModal, {
  i18n,
} from 'ee/oncall_schedules/components/rotations/components/add_edit_rotation_modal.vue';
import { addRotationModalId } from 'ee/oncall_schedules/constants';
import createOncallScheduleRotationMutation from 'ee/oncall_schedules/graphql/mutations/create_oncall_schedule_rotation.mutation.graphql';
import getOncallSchedulesWithRotationsQuery from 'ee/oncall_schedules/graphql/queries/get_oncall_schedules.query.graphql';
import createMockApollo from 'helpers/mock_apollo_helper';
import waitForPromises from 'helpers/wait_for_promises';
import { RENDER_ALL_SLOTS_TEMPLATE, stubComponent } from 'helpers/stub_component';
import searchProjectMembersQuery from '~/graphql_shared/queries/project_user_members_search.query.graphql';
import {
  getOncallSchedulesQueryResponse,
  createRotationResponse,
  createRotationResponseWithErrors,
} from '../../mocks/apollo_mock';
import mockRotation from '../../mocks/mock_rotation.json';

jest.mock('~/alert');
jest.mock('~/lib/utils/color_utils');

const schedule =
  getOncallSchedulesQueryResponse.data.project.incidentManagementOncallSchedules.nodes[0];
const projectPath = 'group/project';

describe('AddEditRotationModal', () => {
  let wrapper;
  let mockApollo;
  let userSearchQueryHandler;
  let createRotationHandler;

  const mockHideModal = jest.fn(function hide() {
    this.$emit('hide');
  });

  const findModal = () => wrapper.findComponent(GlModal);
  const findAlert = () => wrapper.findComponent(GlAlert);
  const findForm = () => wrapper.findComponent(AddEditRotationForm);

  const updateRotationForm = (type, value) => {
    return findForm().vm.$emit('update-rotation-form', {
      type,
      value,
    });
  };

  const createComponent = ({
    rotationHandler = jest.fn().mockResolvedValue(createRotationResponse),
    props = {},
  } = {}) => {
    createRotationHandler = rotationHandler;
    Vue.use(VueApollo);

    mockApollo = createMockApollo([
      [
        getOncallSchedulesWithRotationsQuery,
        jest.fn().mockResolvedValue(getOncallSchedulesQueryResponse),
      ],
      [searchProjectMembersQuery, userSearchQueryHandler],
      [createOncallScheduleRotationMutation, createRotationHandler],
    ]);

    mockApollo.clients.defaultClient.cache.writeQuery({
      query: getOncallSchedulesWithRotationsQuery,
      variables: {
        projectPath,
      },
      data: getOncallSchedulesQueryResponse.data,
    });

    wrapper = shallowMount(AddEditRotationModal, {
      apolloProvider: mockApollo,
      provide: {
        projectPath,
      },
      propsData: {
        modalId: addRotationModalId,
        schedule,
        rotation: mockRotation[0],
        ...props,
      },
      stubs: {
        GlModal: stubComponent(GlModal, {
          methods: {
            hide: mockHideModal,
          },
          template: RENDER_ALL_SLOTS_TEMPLATE,
        }),
      },
    });
  };

  beforeEach(() => {
    createComponent();
  });

  it('renders rotation modal layout', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correct buttons for modal actions', () => {
    expect(findModal().props('actionCancel').text).toBe(i18n.cancel);
    expect(findModal().props('actionPrimary').text).toBe(i18n.addRotation);
    expect(findModal().props('actionPrimary').attributes).toEqual({
      variant: 'confirm',
      loading: false,
      disabled: true,
    });
  });

  describe('Creating rotation', () => {
    it('shows loading spinner on action button while calling a mutation', async () => {
      updateRotationForm('name', mockRotation[0].name);
      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      await nextTick();

      expect(findModal().props('actionPrimary').attributes.loading).toBe(true);
    });

    it('emits "rotation-updated" event and with the confirmation message', async () => {
      updateRotationForm('name', mockRotation[0].name);
      expect(wrapper.emitted('rotation-updated')).toBeUndefined();

      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      await nextTick();
      await waitForPromises();

      const emittedEvents = wrapper.emitted('rotation-updated');
      const emittedMsg = emittedEvents[0][0];

      expect(emittedEvents).toHaveLength(1);
      expect(emittedMsg).toBe(i18n.rotationCreated);
      expect(findModal().props('actionPrimary').attributes.loading).toBe(false);
    });

    it('displays alert if mutation had a recoverable error', async () => {
      createComponent({
        rotationHandler: jest.fn().mockResolvedValue(createRotationResponseWithErrors),
      });
      updateRotationForm('name', mockRotation[0].name);

      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      await nextTick();
      await waitForPromises();

      expect(findAlert().exists()).toBe(true);
      expect(findAlert().text()).toContain('Houston, we have a problem');
      expect(findForm().props('form').name).toBe(mockRotation[0].name);
    });

    it('calls the `searchProjectMembersQuery` query with the search parameter and project path', async () => {
      userSearchQueryHandler = jest.fn().mockResolvedValue();
      createComponent();
      findForm().vm.$emit('filter-participants', 'root');
      await waitForPromises();
      expect(userSearchQueryHandler).toHaveBeenCalledWith({
        search: 'root',
        fullPath: projectPath,
      });
    });

    describe('Validation', () => {
      describe('name', () => {
        it('is valid when name is NOT empty', () => {
          updateRotationForm('name', '');
          expect(findForm().props('validationState').name).toBe(false);
        });

        it('is NOT valid when name is empty', () => {
          updateRotationForm('name', 'Some value');
          expect(findForm().props('validationState').name).toBe(true);
        });
      });

      describe('participants', () => {
        it('is valid when participants array is NOT empty', () => {
          updateRotationForm('participants', ['user1', 'user2']);
          expect(findForm().props('validationState').participants).toBe(true);
        });

        it('is NOT valid when participants array is empty', () => {
          updateRotationForm('participants', []);
          expect(findForm().props('validationState').participants).toBe(false);
        });
      });

      describe('rotationLength', () => {
        it('is valid when length is more than 0', () => {
          updateRotationForm('rotationLength.length', 4);
          expect(findForm().props('validationState').rotationLength).toBe(true);
        });

        it('is valid NOT when length is less than 0', async () => {
          updateRotationForm('rotationLength.length', -4);
          await nextTick();
          expect(findForm().props('validationState').rotationLength).toBe(false);
        });
      });

      describe('startsAt date', () => {
        it('is valid when date is NOT empty', () => {
          updateRotationForm('startsAt.date', new Date('10/12/2021'));
          expect(findForm().props('validationState').startsAt).toBe(true);
        });

        it('is NOT valid when date is empty', () => {
          updateRotationForm('startsAt.date', null);
          expect(findForm().props('validationState').startsAt).toBe(false);
        });
      });

      describe('endsAt date', () => {
        it('is valid when date is empty', () => {
          updateRotationForm('endsAt.date', null);
          expect(findForm().props('validationState').endsAt).toBe(true);
        });

        it('is valid when start date is smaller then end date', () => {
          updateRotationForm('startsAt.date', new Date('9/11/2021'));
          updateRotationForm('endsAt.date', new Date('10/11/2021'));
          expect(findForm().props('validationState').endsAt).toBe(true);
        });

        it('is invalid when start date is larger then end date', () => {
          updateRotationForm('startsAt.date', new Date('11/11/2021'));
          updateRotationForm('endsAt.date', new Date('10/11/2021'));
          expect(findForm().props('validationState').endsAt).toBe(false);
        });

        it('is valid when start and end dates are equal but time is smaller on start date', () => {
          updateRotationForm('startsAt.date', new Date('11/11/2021'));
          updateRotationForm('startsAt.time', 10);
          updateRotationForm('endsAt.date', new Date('11/11/2021'));
          updateRotationForm('endsAt.time', 22);
          expect(findForm().props('validationState').endsAt).toBe(true);
        });

        it('is invalid when start and end dates are equal but time is larger on start date', () => {
          updateRotationForm('startsAt.date', new Date('11/11/2021'));
          updateRotationForm('startsAt.time', 10);
          updateRotationForm('endsAt.date', new Date('11/11/2021'));
          updateRotationForm('endsAt.time', 8);
          expect(findForm().props('validationState').endsAt).toBe(false);
        });
      });

      describe('Toggle primary button state', () => {
        it('should disable primary button when any of the fields is invalid', async () => {
          updateRotationForm('name', 'lalal');
          await nextTick();
          expect(findModal().props('actionPrimary').attributes).toEqual(
            expect.objectContaining({ disabled: true }),
          );
        });

        it('should enable primary button when all fields are valid', async () => {
          updateRotationForm('name', 'Value');
          updateRotationForm('participants', [1, 2, 3]);
          updateRotationForm('startsAt.date', new Date('11/11/2021'));
          updateRotationForm('endsAt.date', new Date('12/10/2021'));
          await nextTick();
          expect(findModal().props('actionPrimary').attributes).toEqual(
            expect.objectContaining({ disabled: false }),
          );
        });
      });
    });
  });

  describe('When editing exisiting rotation', () => {
    beforeEach(async () => {
      await createComponent({ props: { isEditMode: true } });
      await waitForPromises();

      findModal().vm.$emit('show');
    });

    it('should load name correctly', () => {
      expect(findForm().props('form')).toMatchObject({
        name: 'Rotation 242',
      });
    });

    it('should load rotation length correctly', () => {
      expect(findForm().props('form')).toMatchObject({
        rotationLength: {
          length: 2,
          unit: 'WEEKS',
        },
      });
    });

    it('should load participants correctly', () => {
      expect(cloneDeep(findForm().props('form'))).toMatchObject({
        participants: [
          {
            id: 'gid://gitlab/IncidentManagement::OncallParticipant/49',
            username: 'nora.schaden',
            avatarUrl: '/url',
            name: 'nora',
          },
          {
            id: 'gid://gitlab/User/2',
            username: 'racheal.loving',
            avatarUrl: '/url',
            name: 'racheal',
          },
        ],
      });
    });

    it('should load startTime correctly', () => {
      expect(findForm().props('form')).toMatchObject({
        startsAt: {
          date: new Date('2021-01-13T00:00:00.000Z'),
          time: 3,
        },
      });
    });

    it('should load endTime correctly', () => {
      expect(findForm().props('form')).toMatchObject({
        endsAt: {
          date: new Date('2021-03-13T00:00:00.000Z'),
          time: 7,
        },
      });
    });

    it('should load rotation restriction data successfully', () => {
      expect(findForm().props('form')).toMatchObject({
        isRestrictedToTime: true,
        restrictedTo: { startTime: 2, endTime: 10 },
      });
    });
  });
});
