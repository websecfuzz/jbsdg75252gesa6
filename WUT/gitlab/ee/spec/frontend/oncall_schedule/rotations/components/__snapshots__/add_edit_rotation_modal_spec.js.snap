// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditRotationModal renders rotation modal layout 1`] = `
<div>
  <div
    data-testid="slot-default"
  >
    <gl-alert-stub
      dismissible="true"
      dismisslabel="Dismiss"
      politeness="polite"
      primarybuttonlink=""
      primarybuttontext=""
      secondarybuttonlink=""
      secondarybuttontext=""
      showicon="true"
      title=""
      variant="danger"
    >
      Error: Unexpected error whilst calling request handler: value is not a function
    </gl-alert-stub>
    <add-edit-rotation-form-stub
      form="[object Object]"
      participants=""
      schedule="[object Object]"
      validationstate="[object Object]"
    />
  </div>
</div>
`;
